src/pose_estimating.h

```diff
@@ -0,0 +1,162 @@
+/*
+ * @file pose.h
+ * @brief Interface for representation of user's pose
+
+ * @copyright Copyright (C) 2021  Conor Begley
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef SRC_POSE_ESTIMATING_H_
+#define SRC_POSE_ESTIMATING_H_
+
+#include "post_processor.h"
+
+namespace PoseEstimating {
+
+/**
+ * @brief An enum for each important pose body part
+ */
+enum Joint {
+    Head,
+    Neck,
+    Shoulder,
+    Hip,
+    Knee,
+    Foot
+}
+
+/*
+ * @brief A representation of a body part
+ *
+ * Each body part is represented as a `Pose::ConnectedJoint` to record the joints relative position and the connected joints to this joint
+ * 
+ */
+struct ConnectedJoint {
+    Joint joint;
+    /* @brief relative x co-ordinate of joint */
+    float x;
+    /* @brief relative y co-ordinate of joint */
+    float y;
```

Comment:
Could we use a coordinate structure here (similar to the output of post_processor)? It might make things too nested, but I think it might be worth using some more strict types here

Reply:
You can tell you've been studying rust 😂 I think at some point it will still be a float? Whether the user does that here or in the co-ordinate struct it's still open to the same abuse? I don't see the benefit.